metadata-ingestion/src/datahub/ingestion/source/dbt.py

```diff
@@ -1026,9 +1016,18 @@ def _aggregate_owners(
     ) -> List[OwnerClass]:
         owner_list: List[OwnerClass] = []
         if node.owner:
+            owner = node.owner
+            if self.config.owner_naming_pattern:
+                found = re.search(re.compile(self.config.owner_naming_pattern), owner)
+                if found:
+                    owner = found.group(2)
```

Comment:
Hi @arunvasudevan, this is making an assumption that (1) there are at least 2 parenthesized groups in the user provided owner naming pattern and (2) the second one happens to be the owner. If these 2 conditions are not met, we will either get an IndexError or an incorrect value for the owner.

My recommendation is to use (?P\<name\>...) syntax, where name=owner, and use found.group('owner') to extract the owner. The documentation for the pattern also needs to reflect the same. Please refer to [Match Objects documentation](https://docs.python.org/3/library/re.html#match-objects) for details.

Reply:
@rslanka Updated code changes based on the suggestion. Take a look and let me know your comments

Reply:
@rslanka All checks passed as well. Let me know your thoughts